Core/include/Acts/Utilities/ProtoAxis.hpp

```diff
@@ -0,0 +1,134 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Utilities/Axis.hpp"
+#include "Acts/Utilities/AxisDefinitions.hpp"
+#include "Acts/Utilities/Grid.hpp"
+#include "Acts/Utilities/IAxis.hpp"
+
+namespace Acts {
+/// @brief Description of a ProtoAxis which holds an IAxis
+/// and lets the user to define a certain axis type, boundary type
+/// and associated binning.
+///
+/// The IAxis allows via the visitor pattern to access the actual axis type
+/// which helps to create grid creation code by the compiler as done
+/// in the makeGrid helper functions.
+///
+/// In addition to a simple axis definitions, it holds also a description
+/// of the axis direction.
+class ProtoAxis {
+ public:
+  /// Convenience constructors - for variable binning
+  ///
+  /// @param aDir the value/cast in which this is binned
+  /// @param abType the axis boundary type
+  /// @param edges the bin edges (variable binning)
+  ProtoAxis(AxisDirection aDir, Acts::AxisBoundaryType abType,
+            const std::vector<double>& edges);
+
+  /// Convenience constructors - for equidistant binning
+  ///
+  /// @param aDir the value/cast in which this is binned
+  /// @param abType the axis boundary type
+  /// @param minE the lowest edge of the binning
+  /// @param maxE the highest edge of the binning
+  /// @param nbins the number of bins
+  ProtoAxis(AxisDirection aDir, AxisBoundaryType abType, double minE,
+            double maxE, std::size_t nbins);
+
+  /// Placeholder constructor for auto-range binning
+  ///
+  /// @param aDir the value/cast in which this is binned
+  /// @param abType the axis boundary type
+  /// @param nbins the number of bins
+  ///
+  /// @note that auto-range is only supported for equidistant binning
+  ProtoAxis(AxisDirection aDir, AxisBoundaryType abType, std::size_t nbins);
+
+  ProtoAxis(const ProtoAxis&) = delete;
+  ProtoAxis& operator=(const ProtoAxis&) = delete;
+  ProtoAxis(ProtoAxis&&) = default;
+  ProtoAxis& operator=(ProtoAxis&&) = default;
```

Comment:
This is presumably due to the `unique_ptr`? I guess we could consider making `IAxis` copyable, and then modify the copy and assignment operators in `ProtoAxis` to copy the object itself, rather than the `unique_prt`.